src/components/grid_cell_icon/grid_cell_icon.ts

```diff
@@ -0,0 +1,112 @@
+import { Component, ComponentConstructor } from "@odoo/owl";
+import { DEFAULT_VERTICAL_ALIGN, GRID_ICON_EDGE_LENGTH, GRID_ICON_MARGIN } from "../../constants";
+import { positionToZone } from "../../helpers";
+import {
+  Align,
+  CellPosition,
+  DOMCoordinates,
+  SpreadsheetChildEnv,
+  VerticalAlign,
+} from "../../types";
+import { css, cssPropertiesToCss } from "../helpers";
+
+const CSS = css/* scss */ `
+  .o-grid-cell-icon {
+    width: ${GRID_ICON_EDGE_LENGTH}px;
+    height: ${GRID_ICON_EDGE_LENGTH}px;
+  }
+`;
+
+export interface GridCellIconProps {
+  cellPosition: CellPosition;
+  component: ComponentConstructor;
+  horizontalAlign?: Align;
+  defaultHorizontalAlign?: Align;
+  verticalAlign?: VerticalAlign;
+  defaultVerticalAlign?: VerticalAlign;
+  offset?: DOMCoordinates;
+}
+
+export class GridCellIcon extends Component<GridCellIconProps, SpreadsheetChildEnv> {
+  static style = CSS;
+  static template = "o-spreadsheet-GridCellIcon";
+
+  get iconStyle() {
+    const x = this.getIconHorizontalPosition();
+    const y = this.getIconVerticalPosition();
+    return cssPropertiesToCss({
+      top: `${y + (this.props.offset?.y || 0)}px`,
+      left: `${x + (this.props.offset?.x || 0)}px`,
+    });
+  }
+
+  private getIconVerticalPosition(): number {
+    const { sheetId, row } = this.props.cellPosition;
+    const merge = this.env.model.getters.getMerge(this.props.cellPosition);
+    const start = this.env.model.getters.getRowDimensionsInViewport(sheetId, row).start;
+    const end = this.env.model.getters.getRowDimensionsInViewport(
+      sheetId,
+      merge ? merge.bottom : row
+    ).end;
+    const cell = this.env.model.getters.getCell(this.props.cellPosition);
```

Comment:
All the computations for merge, start, end and centeringOffset should be done in the switch/default

Reply:
I agree for `centeringOffset `; merge and end not really since they are used in 2 cases of the switch. And if I leave the `end` computation outside, I might as well leave the start at the same place. It's prettier, and not like it impact the perfs in a meaningful way.